### PR TITLE
Atomic swingset upgrade actions

### LINKED ISSUE #8279: Run upgrade plan to completion during upgrade block
## What is the Problem Being Solved?

#8112 added support for running swingset work derived from the upgrade info contained in the governance proposal. However it did so by piggy backing on the the existing high priority queue, which means that action will likely not complete (or even start) before the end of the upgrade block.

Software upgrades are a great time to perform long expensive operations atomically instead of spreading them over multiple blocks like during normal execution. Any action triggered by an upgrade plan should run to completion during the upgrade block.

## Description of the Design

Instead of pushing an action to the `highPriorityQueue`, we should instead:

- Push to an in memory only "run now" queue
  - Only if the block needs execution (hangover handling)
- In end block check if we have anything in this run now queue and if so
  - run any remaining swingset to quiescence
  - run the run now actions to quiescence (unlimited run policy)
  - empty the run now queue
  - run nothing else, not even a timer poll

## Security Considerations

If the remaining swingset work, or the core proposal work does not get quiescent, the upgrade block may never finish. However that is a degenerate issue of #7938 and can be handled by the same mechanism

## Scaling Considerations

If a "run now" action is performed outside of an upgrade block, it can interact negatively with cosmos consensus. While we do not have immediate plans to use this mechanism outside an upgrade block, it may be a good way to restart all vats before a software upgrade that will upgrade XS (see #7855)

## Test Plan

Docker update test with a check that immediately after the upgrade block the core proposal is in effect (e.g. check a vat incarnation)

## Upgrade Considerations

Yes

### PR DESCRIPTION
**Pull request by mhofman**
closes: #8279

## Description

Builds on #8112 to run the core proposals part of an upgrade to completion within the upgrade block.

### Security Considerations

With an infinite meter, there is a risk that a buggy core proposal or the remaining Swingset work executed before may never run to completion and effectively stall the chain.

### Scaling Considerations

This will cause all upgrade work to happen in the upgrade block, which may take arbitrarily long.

### Documentation Considerations

None

### Testing Considerations

Updated the docker test to verify the network vat is available as soon as the pre test starts, however the docker test infra does not make it possible to test this immediately once the upgrade block is committed, so really it's a best effort.

### Upgrade Considerations

Yes

### CODE REVIEW

**Comment by mhofman** [comment]
Best reviewed commit-by-commit and white space hidden

**Comment by michaelfig** [comment]
I'll have a look at this tomorrow.  When you're up-and-about, can you PTAL at https://github.com/Agoric/agoric-sdk/actions/runs/6032610362/job/16368096857?pr=8287#step:5:25 ?

```
2023-08-31T03:20:37.830Z fake-chain: Cannot boot sim chain: (TypeError#1)
TypeError#1: Cannot destructure property 'beans_per_unit' of 'params' as it is undefined.

  at parseParams (packages/cosmic-swingset/src/params.js:35:21)
  at doBlockingSend (packages/cosmic-swingset/src/launch-chain.js:745:40)
```

**Comment by mhofman** [comment]
Yeah I missed an update to synthetic blocking sends created by simchain. The real chain will have params on the bootstrap block, so the sim chain should too.

File: packages/deployment/upgrade-test/Dockerfile [hunk 1]
```diff
@@ -95,7 +95,7 @@ ARG DEST_IMAGE
 FROM ghcr.io/agoric/agoric-sdk:35 as propose-agoric-upgrade-11
 # This default UPGRADE_INFO_11 is to test core proposals like the network vat.
 # TODO: Maybe replace with a Zoe core proposal, or remove when other paths test it.
```
Comment (michaelfig) [inline packages/deployment/upgrade-test/Dockerfile]: Please also remove these stale comments.
```suggestion
```

**Review by michaelfig — APPROVED** [review]
Looking good!  Thanks for sorting out the prioritization that I balked at. :)

**Pull request merged**